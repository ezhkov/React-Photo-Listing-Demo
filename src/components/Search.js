import React, { Component } from 'react';
import {FormInline, Fa} from 'mdbreact';

export class Search extends Component {

    state = {};


    render() {
        return (
            <div class="flex-container">
                <FormInline className="md-form ml-3">
                    <Fa icon="search" />
                    <input className="form-control form-control-sm ml-3 w-75" type="text" placeholder="Search" aria-label="Search"/>
                </FormInline>
            </div>
        );
    }

}