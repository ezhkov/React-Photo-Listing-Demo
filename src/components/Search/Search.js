import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  FormInline,
  Fa,
} from 'mdbreact';

class Search extends Component {
  constructor(...args) {
    super(...args);
    this.state = {
      inputValue: '',
    };
  }

  submitSearch = (e) => {
    if(e.charCode === 13) {
      e.preventDefault();
      if(this.state.inputValue) this.props.getSearchInputValue(this.state.inputValue);
    } 
  }

  changeInputValue = (e) => {
    this.setState({
      inputValue: e.target.value
    });
    if(this.state.inputValue) this.props.onChangeInputValue(this.state.inputValue);
  };

  render() {
    return (
      <div className="flex-container">
        <FormInline className="md-form ml-3">
          <Fa icon="search" onClick={this.submitSearch} />
          <input onChange={this.changeInputValue} onInput={this.changeInputValue} onKeyPress={this.submitSearch} className="form-control form-control-sm ml-3 w-75" type="text" placeholder="Search" aria-label="Search" value={this.state.inputValue} />
        </FormInline>
      </div>
    );
  }
}

Search.propTypes = {
  getSearchInputValue: PropTypes.func,
};
Search.defaultProps = {
  getSearchInputValue: () => {},
};

export default Search;
