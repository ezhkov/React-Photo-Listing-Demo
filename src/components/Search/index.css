.search {
    margin: 0px 0 30px;
}