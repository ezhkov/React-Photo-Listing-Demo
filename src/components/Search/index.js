import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Input } from 'antd';
import './index.css';
const InputSearch = Input.Search;

class Search extends Component {
  constructor(...args) {
    super(...args);
    this.state = {
      inputValue: this.props.queryText,
    };
  }
  componentDidUpdate = (prevProps, prevState) => {
    if (prevProps.queryText !== this.props.queryText) {
      this.setState({
        inputValue: this.props.queryText,
      });
    }
  };

  submitSearch = () => {
    if(this.state.inputValue) this.props.onSearchInputValue(this.state.inputValue);
  }

  changeInputValue = (e) => {
    this.setState({
      inputValue: e.target.value,
    }, () => {
      this.props.onChangeInputValue(this.state.inputValue);
    });
  };

  render() {
    const { inputValue } = this.state
    return (
      <div className="search">
        <InputSearch
          value={inputValue}
          placeholder="Wallpapers"
          onChange={value => this.changeInputValue(value)}
          onPressEnter={this.submitSearch}
          onSearch={this.submitSearch}
          style={{ width: '100%' }}
        />
      </div>
    );
  }
}

Search.propTypes = {
  queryText: PropTypes.string,
  getSearchInputValue: PropTypes.func,
};
Search.defaultProps = {
  queryText: '',
  getSearchInputValue: () => {},
};

export default Search;
