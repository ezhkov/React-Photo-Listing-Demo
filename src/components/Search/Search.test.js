import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Search from '.'; 

Enzyme.configure({ adapter: new Adapter() });

describe('Test of component of Search', () => {

    it('Test Search placeholder', () => { 

        const search = shallow(<Search />);
        expect(search.find('.search__input').prop('placeholder')).toEqual('Wallpapers');

    });  

    it('Test Search value', () => { 

        const search = shallow(<Search />);
        search.setState({
            inputValue: 'wolf',
        })
        expect(search.find('.search__input').prop('value')).toEqual('wolf');

    });  

    it('Test Snapshot', () => { 
        const search = shallow(<Search />);
        expect(search).toMatchSnapshot();
    });
});