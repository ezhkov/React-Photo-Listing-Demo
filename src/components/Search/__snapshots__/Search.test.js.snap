// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component of Search Test Snapshot 1`] = `ShallowWrapper {}`;
